app/src/main/java/io/gnosis/safe/notifications/NotificationRepository.kt

```diff
@@ -224,9 +229,24 @@ class NotificationRepository(
             }
             .onFailure {
                 registrationUpdateFailed = true
+                handleCloudMessagingTokenIsLinkedToAnotherDeviceError(it)
             }
     }
 
+    private suspend fun handleCloudMessagingTokenIsLinkedToAnotherDeviceError(throwable: Throwable) {
+        val error = throwable.toError()
+        if (error == Error.ErrorCloudMessagingTokenIsLinkedToAnotherDevice) {
+            resetFirebaseToken()
+        }
+    }
+
+    private suspend fun resetFirebaseToken() {
+        kotlin.runCatching {
+            FirebaseInstanceId.getInstance().deleteInstanceId()
+            FirebaseInstanceId.getInstance().deleteToken(getCloudMessagingToken()!!, INSTANCE_ID_SCOPE)
+        }
+    }
+
     suspend fun unregister() {
         kotlin.runCatching {
             deviceUuid?.let {
```

Comment:
we should delete deviceUuid after successful unregister request

Reply:
About `unregister()`. It is never called. Maybe we can delete it. Who should call it and under what circumstances?

Reply:
I think it should be called when the last safe is removed from the app. But then the app is quite useless.